src/components/dropdown/dropdown-button/dropdown-button.vue

```diff
@@ -0,0 +1,117 @@
+<script setup lang="ts">
+import { computed, onMounted, ref, type Ref } from 'vue';
+import FoamButton from '@/components/button/button.vue';
+import { v4 as uuidv4 } from 'uuid';
+import { createPopper } from '@popperjs/core';
+import { useRemoveRecordedStroke } from '@/composables/tabTrap'
+import useDetectOutsideClick from '@/composables/clickOutsideComponent'
+
+type DropdownProps = {
+    variant?: string
+    text?: string
+    icon?: string | null
+    width?: number | null
+    align?: string
+    disabled?: boolean
+};
+
+const props = withDefaults(defineProps<DropdownProps>(), {
+    variant: 'secondary',
+    text: 'Foam Dropdown',
+    icon: null,
+    width: null,
+    align: 'end',
+    disabled: false
+});
+
+const showDropdown: Ref = ref<boolean>(false);
+const dropdownId = ref<string>('dropdown-' + uuidv4())
+const buttonId = ref<string>('btn-' + uuidv4())
+const dropdown: Ref = ref()
+const dropdownAlign = ref<string>(props.align)
+
+onMounted(() => {
+  if(props.align == 'end' || props.align == 'start') {
+    return
+  } else {
+    dropdownAlign.value = 'end'
+  }  
+})
+
+const popperInstance = computed(() => {
+  const buttonElem = document.querySelector(`#${buttonId.value}`) as HTMLElement
+  const dropdownElem = document.querySelector(`.${dropdownId.value}`) as HTMLElement
+  return createPopper(buttonElem, dropdownElem, {
+    placement: 'auto-start',
+    modifiers: [
+      {
+        name: 'flip',
+        options: {
+          allowedAutoPlacements: [`bottom-${dropdownAlign.value}`, `top-${dropdownAlign.value}`]
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 2],
+        }
+      }
+    ],
+    strategy: 'absolute'
+  });
+})
+
+const handleShowDropdown: () => void = () => {
+  showDropdown.value = !showDropdown.value
+
+  if(showDropdown.value) {
+    popperInstance.value.update()
+    dropdown.value.setAttribute('data-show', '')
+  } else {   
+    dropdown.value.removeAttribute('data-show')
+  }
+}
+
+const componentRef = ref()
+// Close dropdown on click outside the component
+useDetectOutsideClick(componentRef, () => { 
+  showDropdown.value = false
+  dropdown.value.removeAttribute('data-show')
+})
+</script>
+
+<template>
+  <section
+    :id="dropdownId"
+    ref="componentRef"
+    class="dropdown-container"
+    :data-test="dropdownAlign"
+  >
+    <FoamButton
+      :id="buttonId"
+      class="dropdown-button"
+      dropdown
+      :text="props.text"
+      :variant="props.variant"
+      :active-dropdown="showDropdown"
+      :icon="props.icon"
+      :disabled="props.disabled"
+      @click="handleShowDropdown"
+      @keyup="useRemoveRecordedStroke($event)"
+    />
+    <div
+      id="dropdown"
+      ref="dropdown"
+      :class="dropdownId"
+      class="subtitle2"
+      :style="{ width: props.width ? props.width + 'px !important' : 'fit-content' }"
+      @keyup.escape="handleShowDropdown"
+    >
+      <slot />
```

Comment:
One thing you could think about in the future is to use named-slots. It's easier for the readability and also makes life easier if you decide to add more slots to a component.

Reply:
Alright, even if you 'know' you'll only have one slot. Naming them should be the norm either way?